jobbergate-api/tests/apps/clusters/test_routers.py

```diff
@@ -125,3 +125,34 @@ async def test_get_cluster_status__bad_permission(self, client, inject_security_
         response = await client.get("/jobbergate/clusters/status")
 
         assert response.status_code == status.HTTP_403_FORBIDDEN
+
+
+class TestGetClusterStatus:
+    @pytest.mark.parametrize("permission", (Permissions.ADMIN, Permissions.CLUSTERS_READ))
+    @pytest.mark.parametrize("client_id", ("client-1", "client-2", "client-3"))
+    async def test_get_cluster_status_by_client_id__not_found(
+        self, permission, client_id, client, inject_security_header, synth_session
+    ):
+        inject_security_header("who@cares.com", permission)
+
+        response = await client.get(f"/jobbergate/clusters/status/{client_id}")
+        assert response.status_code == status.HTTP_404_NOT_FOUND
+
+    @pytest.mark.parametrize("permission", (Permissions.ADMIN, Permissions.CLUSTERS_READ))
+    @pytest.mark.parametrize("client_id", ("client-1", "client-2", "client-3"))
```

Comment:
Why parametrize these tests? I don't think it would add any additional verification to repeat the test 3 times.

Reply:
the `client_id` is a variable, so, I think it should be parametrized. Consider the following function for example:
```python
def example(client_id: str):
    return client_id.lower()

Reply:
since the `client_id` is a variable in this case, I do think it should be parametrized since I can write logic to make the test pass based on the input but the function itself doesn't behave as it should

Reply:
Both are good points, I see no need to repeat the tests in this case, the three of them will check the same thing. Client id can still be a variable set on the body:

```python
     async def test_get_cluster_status_by_client_id__found(
         self, permission, client, inject_security_header, synth_session
     ):
         client_id = "dummy-client"
```